smarts/zoo/worker.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import argparse
+import tempfile
+import sys
+import pathlib
+import subprocess
+
+from multiprocessing.connection import Listener
+
+from smarts.core.utils.networking import find_free_port
+
+
+def spawn_networked_agent(port):
+    cmd = [
+        sys.executable,  # path to the current python binary
+        str((pathlib.Path(__file__).parent / "run_agent.py").absolute().resolve()),
+        "--port",
+        str(port),
+    ]
+
+    agent_proc = subprocess.Popen(cmd)
+    return agent_proc
+
+
+def spawn_local_agent(socket_file):
+    cmd = [
+        sys.executable,  # path to the current python binary
+        str((pathlib.Path(__file__).parent / "run_agent.py").absolute().resolve()),
+        "--socket_file",
+        socket_file,
+    ]
+
+    agent_proc = subprocess.Popen(cmd)
+    return agent_proc
+
+
+def handle_request(request):
+    if request == "allocate_networked_agent":
+        port = find_free_port()
+        proc = spawn_networked_agent(port)
+        return proc, {"port": port, "result": "success"}
+
+    elif request == "allocate_local_agent":
+        sock_file = tempfile.mktemp()
+        proc = spawn_local_agent(sock_file)
+        return proc, {"socket_file": sock_file, "result": "success"}
+
+    else:
+        return None, {"result": "error", "msg": "bad request"}
+
+
+def listen(port):
+    logging.basicConfig(level=logging.INFO)
```

Comment:
I think this is not the place for setting the logging level.

Reply:
Moved it to the global scope